mmv1/third_party/terraform/tests/data_source_google_compute_instance_group_manager_test.go

```diff
@@ -0,0 +1,215 @@
+package google
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataSourceGoogleComputeInstanceGroupManager(t *testing.T) {
+	t.Parallel()
+
+	zoneName := "us-central1-a"
+	igmName := "tf-tst-igm" + randString(t, 6)
+
+	context := map[string]interface{}{
+		"zoneName":     zoneName,
+		"igmName":      igmName,
+		"baseName":     "tf-tst-igm-base" + randString(t, 6),
+		"poolName":     "tf-tst-pool" + randString(t, 6),
+		"templateName": "tf-tst-templt" + randString(t, 6),
+		"autoHealName": "tf-tst-ah-name" + randString(t, 6),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceGoogleComputeInstanceGroupManager_basic1(context),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.google_compute_instance_group_manager.data_source", "project", getTestProjectFromEnv()),
+					resource.TestCheckResourceAttr("data.google_compute_instance_group_manager.data_source", "zone", zoneName),
+					resource.TestCheckResourceAttr("data.google_compute_instance_group_manager.data_source", "name", igmName)),
+			},
+			{
+				Config: testAccDataSourceGoogleComputeInstanceGroupManager_basic2(context),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.google_compute_instance_group_manager.data_source", "project", getTestProjectFromEnv()),
+					resource.TestCheckResourceAttr("data.google_compute_instance_group_manager.data_source", "zone", zoneName),
+					resource.TestCheckResourceAttr("data.google_compute_instance_group_manager.data_source", "name", igmName)),
+			},
+		},
+	})
+}
+
+func testAccDataSourceGoogleComputeInstanceGroupManager_basic1(context map[string]interface{}) string {
+	return Nprintf(`
+	resource "google_compute_health_check" "autohealing" {
+		name                = "%{autoHealName}"
+		check_interval_sec  = 5
+		timeout_sec         = 5
+		healthy_threshold   = 2
+		unhealthy_threshold = 10 # 50 seconds
+
+		http_health_check {
+		  request_path = "/healthz"
+		  port         = "8080"
+		}
+	}
+
+	resource "google_compute_instance_group_manager" "appserver" {
+		name = "%{igmName}"
+		base_instance_name = "%{baseName}"
+		zone               = "us-central1-a"
+
+		version {
+		  instance_template  = google_compute_instance_template.igm-basic.id
+		  name = "primary"
+		}
+
+		all_instances_config {
```

Comment:
`all_instances_config` is beta-only feature. Consider remove this from the generate test file or add add version guard around the code that should only be in beta like
```
<% unless version == 'ga' -%>
// test case + config here
<% end -%>
```
Otherwise, test will fail in GA

Reply:
this property has been removed.